src/Database/PG/Query/Connection.hs

```diff
@@ -114,6 +115,11 @@ throwPGConnErr
   :: MonadError PGError m => PGConnErr -> m a
 throwPGConnErr = throwError . Right
 
+readConnErr :: PQ.Connection -> IO DT.Text
+readConnErr conn = do
+  m <- PQ.errorMessage conn
+  return $  maybe "(empty connection error message)" lenientDecodeUtf8 m
```

Comment:
This used to be `maybe mempty ...`; the change should make for more readable/understandable error messages in the rare case that `PQ.errorMessage` return `Nothing` (should probably only happen in that OOM case below, but a bit unclear)